packages/features/ee/workflows/components/WebCallDialog.tsx

```diff
@@ -0,0 +1,424 @@
+        } else {
+          await retellWebClientRef.current.mute();
+        }
+        setIsMuted(!isMuted);
+      } catch (error) {
+        console.error("Error toggling mute:", error);
+      }
+    }
+  };
+
+  const _formatDuration = (seconds: number) => {
+    const mins = Math.floor(seconds / 60);
+    const secs = seconds % 60;
+    return `${mins.toString().padStart(2, "0")}:${secs.toString().padStart(2, "0")}`;
+  };
+
+  const getStatusIcon = () => {
+    switch (callStatus) {
+      case "connecting":
+        return <Icon name="loader" className="h-4 w-4 animate-spin rounded-full" />;
+      case "active":
+        return <Icon name="phone" className="text-success h-4 w-4" />;
+      case "ended":
+        return <Icon name="phone-off" className="text-muted h-4 w-4" />;
+      case "error":
+        return <Icon name="triangle-alert" className="text-error h-4 w-4" />;
+      default:
+        return <Icon name="phone" className="text-subtle h-4 w-4" />;
+    }
+  };
+
+  const getStatusText = () => {
+    switch (callStatus) {
+      case "connecting":
+        return t("connecting_to_agent");
+      case "active":
+        return t("call_active_duration", { duration: _formatDuration(_callDuration) });
+      case "ended":
+        return t("call_ended_duration", { duration: _formatDuration(_callDuration) });
+      case "error":
+        return t("call_error");
+      default:
+        return t("ready_to_start_call");
+    }
+  };
+
+  useEffect(() => {
+    if (isEmbed) {
+      return;
+    }
+
+    if (transcriptEndRef.current) {
+      // eslint-disable-next-line @calcom/eslint/no-scroll-into-view-embed
+      transcriptEndRef.current.scrollIntoView({ behavior: "smooth" });
+    }
+  }, [transcript, isEmbed]);
+
+  useEffect(() => {
+    return () => {
+      stopDurationTimer();
+      if (retellWebClientRef.current && callStatus === "active") {
+        try {
+          const stopCallResult = retellWebClientRef.current.stopCall();
+          if (stopCallResult && typeof stopCallResult.catch === "function") {
+            stopCallResult.catch(console.error);
+          }
+        } catch (error) {
+          console.error("Error stopping call during cleanup:", error);
+        }
+      }
+    };
+  }, [callStatus]);
+
+  const resetDialogState = () => {
+    setCallStatus("idle");
+    setTranscript([]);
+    setIsMuted(false);
+    setCallDuration(0);
+    setError(null);
+    callStartTimeRef.current = null;
+    stopDurationTimer();
+    retellWebClientRef.current = null;
+  };
+
+  const handleClose = () => {
+    if (callStatus === "active") {
+      handleEndCall();
+    }
+    resetDialogState();
+    onOpenChange(false);
+  };
+
+  return (
+    <Dialog
+      open={open}
+      onOpenChange={(open) => {
+        if (!open) {
+          if (callStatus === "active") {
+            handleEndCall();
+          }
+          resetDialogState();
+        }
+        onOpenChange(open);
+      }}>
+      <DialogContent
```

Comment:
At the top of this dialog, we should display a info message that a web call will deduct credits

If the user has no credits, I suggest the following:
- Disable the “Test web call” button and show a tooltip
- At the top, explain that this action requires credits and include a link to the credits page where they can purchase credits